packages/main/src/TimeSelectionClocks.ts

```diff
@@ -355,55 +365,90 @@ class TimeSelectionClocks extends TimePickerInternals {
 				"itemMax": 60,
 				"value": time.seconds,
 				"stringValue": this._seconds,
-				"textValue": `${time.seconds} ${this.secondsLabel}`, // possible concatenation
+				"textValue": `${time.seconds} ${this.secondsLabel}`,
 				"displayStep": 5,
 				"lastItemReplacement": 0,
-				"showInnerCircle": false,
+				"hideFractions": false,
 				"prependZero": false,
 				"hasSeparator": this._entities.length > 0,
 				"active": false,
+				"skipAnimation": false,
 				"attributes": {
 					"min": 0,
 					"max": 59,
 					"step": 1,
 				},
 			});
 		}
+
 		this._entities[this._activeIndex].active = true;
+		this._entities[this._activeIndex].skipAnimation = this._skipAnimation;
 		this._createPeriodComponent();
 	}
 
 	/**
 	 * Switches to the specific clock by name.
 	 * @param clockName the name of the clock
+	 * @param skipAnimation whether to skip transition animation while displaying the next clock
 	 */
-	_switchTo(clockName: string) {
+	_switchTo(clockName: string, skipAnimation = false) {
 		const key = this._componentKey(clockName);
 		if (this._componentMap[key] !== undefined) {
-			this._switchClock(this._componentMap[key]);
+			this._switchClock(this._componentMap[key], skipAnimation);
 		}
 	}
 
 	/**
 	 * Switches to the specific clock by its index in _clocks property.
 	 * @param clockIndex the index of the clock
+	 * @param skipAnimation whether to skip transition animation while displaying the next clock
 	 */
-	_switchClock(clockIndex: number) {
-		const newButton = this._buttonComponent(clockIndex);
+	_switchClock(clockIndex: number, skipAnimation = false) {
+		if (this._activeIndex === clockIndex) {
+			return;
+		}
```

Comment:
we could add one more early return here, to avoid potetial runtime errors which could happen in different frameworks for example, like React, Vue, maybe it would look something like that:

```ts
if (!this._entities.length || clockIndex >= this._entities.length) { 
    return;
}
```

Reply:
Done.